openwisp_users/tests/test_admin.py

```diff
@@ -493,6 +493,44 @@ def test_organization_uuid_field(self):
         html = '<input type="text" name="name" value="default"'
         self.assertNotContains(response, html)
 
+    def test__action_active(self):
+        user = User.objects.create(is_active=False)
+        path = reverse('admin:openwisp_users_user_changelist')
+        self.client.force_login(self._get_admin())
+        post_data = {
+                '_selected_action': [user.pk],
+                'action': 'make_active',
+                'csrfmiddlewaretoken': 'test',
+                'confirmation': 'Confirm',
+            }
+        response = self.client.post(path, post_data, follow=True)
+        self.assertEqual(response.status_code, 200)
```

Comment:
please ensure the user has been flagged as active after the operation:

```python
user.refresh_from_db()
self.assertTrue(user.is_active)
```

Without this final assertion, the whole test method is useless, because the feature can break and the test will pass anyway.

Reply:
@nemesisdesign, thanks for suggesting this, I have implemented it. I just realized even when the response status is 200, it might be that the action was not performed which as you said made the test useless.